lib/linter/linter.js

```diff
@@ -584,10 +586,17 @@ function stripUnicodeBOM(text) {
 /**
  * Get the options for a rule (not including severity), if any
  * @param {Array|number} ruleConfig rule configuration
+ * @param {string} ruleId rule id
+ * @param {Object} rule rule schema
  * @returns {Array} of rule options, empty Array if none
  */
-function getRuleOptions(ruleConfig) {
+function getRuleOptions(ruleConfig, ruleId = "", rule = {}) {
```

Comment:
I agree with [this comment](https://github.com/eslint/eslint/pull/13034#pullrequestreview-386547863). Would it be possible to clone any config files we read from the file system at the time we read them, rather than only cloning `rules` key? Seems like it would be safest to never reference the original config object from within ESLint itself.

Reply:
[here](https://github.com/eslint/eslint/blob/master/lib/cli-engine/config-array-factory.js#L625) ?

or in you want to clone while loading then may be [here](https://github.com/eslint/eslint/blob/master/lib/cli-engine/config-array-factory.js#L493-L496) ?

Reply:
Thanks, this is more along the lines of what I was thinking of. I'm not terribly familiar with this part of our codebase and will have to make sure this covers all of our use cases, but I think [this spot](https://github.com/eslint/eslint/blob/6b7030b1a1e1e3d1a3953cfa9722074d6a6fc1a9/lib/cli-engine/config-array-factory.js#L625) makes sense!

Reply:
done